de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/v1_9/joiner/JoinerNodeModel.java

```diff
@@ -406,20 +404,31 @@ protected void loadJsonSetting() throws IOException, CanceledExecutionException
         visualizationScript = null;
       }
     }
-
+    List<Parameter> newFirstModelParameters =
+        SwaggerUtil.getParameter(firstInputPort.modelMetadata);
+    List<Parameter> newSecondModelParameters =
+        SwaggerUtil.getParameter(secondInputPort.modelMetadata);
     JoinerViewValue viewValue = getViewValue();
     viewValue.joinRelations = nodeSettings.connections;
-    viewValue.modelMetaData = nodeSettings.modelMetaData;
     viewValue.modelScriptTree = sourceTree;
-
     JoinerViewRepresentation representation = getViewRepresentation();
     if (nodeSettings.firstModelParameters != null) {
       representation.setFirstModelParameters(nodeSettings.firstModelParameters);
     }
     if (nodeSettings.secondModelParameters != null) {
       representation.setSecondModelParameters(nodeSettings.secondModelParameters);
     }
+
+
+    if (!JoinerNodeUtil.parametersNeedUpdate(newFirstModelParameters)) {
+      representation.updateParameters(newFirstModelParameters, newSecondModelParameters);
+      viewValue.updateParameters(newFirstModelParameters, newSecondModelParameters);
```

Comment:
Do not we need to set the viewValue.modelMetadata when the parameters need to be updated? What happens to the metadata in this case?

Reply:
in the case that the parameters need to be updated the metadata will be generated again and will not be taken from saved setting.